File: tests/unit/src/main/scala/tests/CompileResult.scala
package tests

import org.scip_code.scip.Document
import org.scip_code.scip.Index

case class CompileResult(
    byteCode: Array[Byte],
    stdout: String,
    documents: Seq[Document],
    isSuccess: Boolean
) {
  def document: Option[Document] = documents.headOption

  def merge(other: CompileResult): CompileResult = copy(
    byteCode = this.byteCode ++ other.byteCode,
    stdout = this.stdout ++ other.stdout,
    documents = this.documents ++ other.documents,
    isSuccess = this.isSuccess && other.isSuccess
  )
}

object CompileResult {
  val empty: CompileResult = CompileResult(
    Array.emptyByteArray,
    "",
    Seq.empty,
    isSuccess = true
  )

  /**
   * Convenience: builds an empty SCIP {@link Index} from this result's
   * documents.
   */
  def index(documents: Seq[Document]): Index = Index
    .newBuilder()
    .addAllDocuments(
      scala.jdk.CollectionConverters.SeqHasAsJava(documents).asJava
    )
    .build()
}

File: tests/unit/src/main/java/tests/CompileResult.java
package tests;

import java.util.Collections;
import java.util.List;
import org.scip_code.scip.Document;

/** Result of an in-process javac compilation driven by {@link TestCompiler}. */
public final class CompileResult {
  public final byte[] byteCode;
  public final String stdout;
  public final List<Document> documents;
  public final boolean isSuccess;

  public CompileResult(
      byte[] byteCode, String stdout, List<Document> documents, boolean isSuccess) {
    this.byteCode = byteCode;
    this.stdout = stdout;
    this.documents = documents;
    this.isSuccess = isSuccess;
  }

  public static CompileResult empty() {
    return new CompileResult(new byte[0], "", Collections.emptyList(), true);
  }
}
